### DISCUSSION TITLE
Install of Ansys Rocky fails using "Silent" switch, but not "Interactive'

### DISCUSSION THREAD

**Discussion by padrickr** [PSAppDeployToolkit/PSAppDeployToolkit#1063] [Q&A]
Seeing a strange issue with Ansys Rocky install using PSADT. I know Rocky struggles with System environment variables, so I added logic to set environment variables to a different location for the session, and it works fine. The issue I am encountering is with running the installation with the "Silent" switch. I get the following error toward the end of the install:

![Screenshot 2024-10-11 092150](https://github.com/user-attachments/assets/f6f71486-a037-487e-9a79-e662c049d307)

Yes, it's python which adds extra complications. However, when I run deploy-application.exe (as system) without the Silent switch, I click Continue and it installs without issue.

Clearly there's something different in the session when doing it interactively vs silently -- even when both run as system. I just can't seem to find out what. I scoured procmon logs, regshot, and everything else I could find that would point to an environment variable, file, or registry setting that's user specific during this phase, but can't find anything.

Can anyone tell me why it would fail on silent but not interactive when both are run as system?

**Comment by mjr4077au** [comment]
Can you please test and advise whether it works as SYSTEM but without using something like ServiceUI, etc? It doesn't have some privileges like symbolic link creation which could cause an issue

**Comment by DanGough** [comment]
Also, test running the silent install command directly (without PSADT) via psexec - if it works via `psexec /s /i` but not via `psexec /s` then the installer may have issues running in session zero.
